aten/src/ATen/native/TensorIterator.h

```diff
@@ -399,20 +397,23 @@ struct CAFFE2_API TensorIterator {
   NameVector names_;
   SmallVector<OperandInfo, 4> operands_;
   int num_outputs_ = 0;
-  CommonDTypeStrategy common_dtype_strategy_ = CommonDTypeStrategy::CHECK;
   ScalarType common_dtype_ = ScalarType::Undefined;
   bool has_coalesced_dimensions_ = false;
   bool accumulate_ = false;
   bool resize_outputs_ = true;
   bool is_reduction_ = false;
   bool allow_cpu_scalars_ = false;
-  bool promote_gpu_output_dtypes_ = false;
   bool final_output_ = true;
   bool check_mem_overlap_ = false;
   bool all_ops_same_shape_ = false;
   bool requires_channels_last_output_ = false;
   bool requires_channels_last_3d_output_ = false;
   bool static_shape_ = false;
+  bool check_all_same_dtype_ = true;
+  bool check_all_same_device_ = true;
+  bool promote_inputs_to_common_dtype_ = false;
+  bool cast_common_dtype_to_outputs_ = false;
+  bool enforce_safe_casting_to_output_ = false;
```

Comment:
What is "safe casting to output"?

Reply:
Your plan to make the comment block in TensorIterator.cpp a note and reference it from here should clarify. I agree it's not clear.